api/envoy/service/auth/v2alpha/external_auth.proto

```diff
@@ -24,21 +25,37 @@ message CheckRequest {
   AttributeContext attributes = 1;
 }
 
+// Http attributes for denied responses.
+message DeniedHttpResponse {
+  // Http status code.
+  envoy.api.v2.core.HttpStatus status = 1;
+
+  // Http entity headers.
+  map<string, string> headers = 2;
+
+  // Http entity body.
+  string body = 3;
+}
+
+// Http attributes for ok responses.
+message OkHttpResponse {
+  // Http entity headers.
+  map<string, string> headers = 2;
```

Comment:
NIt: this should be field number `1`. How come you decided to duplicate `headers` rather than factor it out?

Reply:
👍 I'm not sure if I quite understand your question. I decided to separate the content into two distinct messages and then use `oneof` in CheckResponse. Both DeniedHttpResponse & OkHttpResponse need to headers.